### PR TITLE
Generic data reader

### PR DESCRIPTION
**Pull request by humleflue**
Also moved the ICsvLineParser interface back to the DataReader

### CODE REVIEW

File: DataReader/ICsvLineParser.cs [hunk 1]
```diff
@@ -1,18 +1,19 @@
...
     /// The job of a line-parser class is to parse a line of csv-data into the wanted format.
     /// </summary>
-    interface ICsvLineParser
+    public interface ICsvLineParser
     {
         CultureInfo CultureInfo { get; }
 
         /// <summary>
-        /// Should parse the Line property to an IAssetRecord according to the provided parserFunc
+        /// Should return a
```
Comment (Pattrigue) [inline DataReader/ICsvLineParser.cs:16]: Incomplete documentation

**Review by Pattrigue — CHANGES_REQUESTED** [review]
Please fix missing documentation

**Review by Ivikhostrup — APPROVED** [review]
Queen Elizabeth approves

**Comment by humleflue** [comment]
Deprecated. We prefer using LINQ instead. See pr https://github.com/chhoumann/p3/pull/49

**Pull request closed without merging**